deluge/log.py

```diff
@@ -125,13 +128,10 @@ def setup_logger(level="error", filename=None, filemode="w"):
 
     root_logger = logging.getLogger()
 
-    if filename and filemode == "a":
+    if filename and logrotate:
         handler = logging.handlers.RotatingFileHandler(
-            filename, filemode,
-            maxBytes=50 * 1024 * 1024,  # 50 Mb
-            backupCount=3,
-            encoding="utf-8",
-            delay=0
+            filename, filemode, maxBytes=logrotate,
```

Comment:
Looking at docs filemode must be ignored as it's a named arg `mode` and defaults to `a` anyway so can be omitted.